modules/System Admin/alarmProcess.php

```diff
@@ -57,88 +58,35 @@
-            $result = $connection2->prepare($sql);
-            $result->execute($data);
-        } catch (PDOException $e) {
-            $checkFail = true;
-        }
-
+        $alarmTest = $alarmGateway->selectBy(['status' => 'Current'])->fetch();
+        
         //Alarm is being turned on, so insert new record
         if ($alarm == 'General' or $alarm == 'Lockdown' or $alarm == 'Custom') {
-            if ($checkFail == true) {
-                $fail = true;
+            if (empty($alarmTest)) {
+                //Write alarm to database
+                $data = array('type' => $alarm, 'status' => 'Current', 'gibbonPersonID' => $gibbon->session->get('gibbonPersonID'), 'timestampStart' => date('Y-m-d H:i:s'));
+                $alarmGateway->insert($data);    
             } else {
-                if ($result->rowCount() == 0) {
-                    //Write alarm to database
-                    try {
-                        $data = array('type' => $alarm, 'gibbonPersonID' => $_SESSION[$guid]['gibbonPersonID'], 'timestampStart' => date('Y-m-d H:i:s'));
-                        $sql = "INSERT INTO gibbonAlarm SET type=:type, status='Current', gibbonPersonID=:gibbonPersonID, timestampStart=:timestampStart";
-                        $result = $connection2->prepare($sql);
-                        $result->execute($data);
-                    } catch (PDOException $e) {
-                        $fail = true;
-                    }
-                } else {
-                    $row = $result->fetch();
-                    try {
-                        $data = array('type' => $alarm, 'gibbonAlarmID' => $row['gibbonAlarmID']);
-                        $sql = 'UPDATE gibbonAlarm SET type=:type WHERE gibbonAlarmID=:gibbonAlarmID';
-                        $result = $connection2->prepare($sql);
-                        $result->execute($data);
-                    } catch (PDOException $e) {
-                        $fail = true;
-                    }
-                }
+                $alarmGateway->updateWhere(['gibbonAlarmID' => $alarmTest['gibbonAlarmID']], ['type' => $alarm]);
             }
         } elseif ($alarmCurrent != $alarm) {
-            if ($result->rowCount() == 1) {
-                $row = $result->fetch();
-                try {
-                    $data = array('timestampEnd' => date('Y-m-d H:i:s'), 'gibbonAlarmID' => $row['gibbonAlarmID']);
-                    $sql = "UPDATE gibbonAlarm SET status='Past', timestampEnd=:timestampEnd WHERE gibbonAlarmID=:gibbonAlarmID";
-                    $result = $connection2->prepare($sql);
-                    $result->execute($data);
-                } catch (PDOException $e) {
-                    $fail = true;
-                }
-            } else {
-                $fail = true;
-            }
+            $alarmGateway->updateWhere(['gibbonAlarmID' => $alarmTest['gibbonAlarmID']], ['status' => 'Past', 'timestampEnd' => date('Y-m-d H:i:s')]);
```

Comment:
The basic update method works well for a Gateway's primary key, and is a little less verbose, eg:
```suggestion
            $alarmGateway->update($alarmTest['gibbonAlarmID'], ['status' => 'Past', 'timestampEnd' => date('Y-m-d H:i:s')]);
```